pkg/manifests/manifests.go

```diff
@@ -953,6 +953,17 @@ func (f *Factory) NodeExporterServiceMonitor() (*monv1.ServiceMonitor, error) {
 	return sm, nil
 }
 
+func (f *Factory) updateNodeExporterArgs(args []string) []string {
+	if f.config.ClusterMonitoringConfiguration.NodeExporterConfig.Collectors.CpuFreq.Enabled {
+		args = unsetArg(args, "--no-collector.cpufreq")
+		args = setArg(args, "--collector.cpufreq", "")
+	} else {
+		args = unsetArg(args, "--collector.cpufreq")
+		args = setArg(args, "--no-collector.cpufreq", "")
+	}
```

Comment:
Sorry if I wasn't clear in my previous comments but I'd simplify further now that the static asset doesn't enable/disable explicitly the cpufreq collector. This change also means that we don't need the unsetArg function.

```suggestion
	if f.config.ClusterMonitoringConfiguration.NodeExporterConfig.Collectors.CpuFreq.Enabled {
		args = setArg(args, "--collector.cpufreq", "")
	} else {
		args = setArg(args, "--no-collector.cpufreq", "")
	}
```